test/functorch/test_control_flow.py

```diff
@@ -1174,6 +1203,118 @@ def fwbw(map_op, f, x, y):
         fake_outs = fwbw(_fake_map, f, x, y)
         self.assertEqual(true_outs, fake_outs)
 
+    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    def test_generic_associative_scan_vmap(self):
+        device = torch.device("cuda")
+
+        def add(x: torch.Tensor, y: torch.Tensor):
+            return x + y
+
+        x = torch.tile(
+            torch.unsqueeze(
+                torch.unsqueeze(
+                    torch.arange(
+                        0, 10, device=device, dtype=torch.float32, requires_grad=True
+                    ),
+                    0,
+                ),
+                -1,
+            ),
+            (4, 1, 4),
+        )
+        torch.compiler.reset()
+
+        def associative_scan_fct(x):
+            return associative_scan(add, x, 0)
+
+        with torch._dynamo.utils.disable_cache_limit():
+            associative_scan1 = torch.compile(
+                torch.vmap(associative_scan_fct, in_dims=0), fullgraph=True
+            )
+            associative_scan2 = torch.vmap(associative_scan_fct, in_dims=0)
```

Comment:
Can we add make a test that varies in_dims?

Reply:
I've added a new testcase for that